skyportal/handlers/api/group.py

```diff
@@ -381,49 +379,37 @@ def post(self, group_id, *ignored_args):
             return self.error("Cannot add users to single-user groups.")
         user = User.query.filter(User.username == username.lower()).first()
         if user is None:
-            groups = Group.query.filter(Group.id == group_id).all()
-            if cfg["invitations.enabled"]:
-                invite_token = str(uuid.uuid4())
-                DBSession.add(
-                    Invitation(
-                        token=invite_token,
-                        groups=groups,
-                        admin_for_groups=[admin],
-                        user_email=username,
-                        invited_by=self.associated_user_object,
-                    )
-                )
-                user_id = None
-            else:
-                user_id = add_user_and_setup_groups(
-                    username=username,
-                    roles=["Full user"],
-                    group_ids_and_admin=[[group_id, admin]],
-                )
+            user_id = add_user_and_setup_groups(
+                username=username,
+                roles=["Full user"],
+                group_ids_and_admin=[[group_id, admin]],
+            )
         else:
             user_id = user.id
-            # Just add new GroupUser
+            # Ensure user has sufficient stream access to be added to group
+            if group.streams:
+                user_stream_ids = [
+                    su.stream_id
+                    for su in DBSession()
+                    .query(StreamUser)
+                    .filter(StreamUser.user_id == user.id)
+                    .all()
+                ]
+                if not all([stream.id in user_stream_ids for stream in group.streams]):
```

Comment:
The culmination of months of thinking about ACLs 😂

Reply:
:+1: :joy: